mono/metadata/icall.c

```diff
@@ -446,6 +538,17 @@ array_set_value_impl (MonoArrayHandle arr_handle, MonoObjectHandle value_handle,
 	if (vt_isenum)
 		vt = mono_class_enum_basetype_internal (m_class_get_byval_arg (vc)->data.klass)->type;
 
+#if ENABLE_NETCORE
+	// Treat MONO_TYPE_U/I as MONO_TYPE_U8/I8/U4/I4
+#if SIZEOF_VOID_P == 8
```

Comment:
This seems like a recurring pattern in mono to capture somehow, the mapping of `I4` or `I8` to native `I`, and `U`.

Reply:
I tried to find any similar patterns but with no luck